File: test-suite-docs-java/src/test/java/io/micronaut/mcp/docs/tools/fetch/MicronautModulesFetchTest.java
package io.micronaut.mcp.docs.tools.fetch;

import io.micronaut.context.annotation.Property;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.BlockingHttpClient;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.mcp.server.tools.fetch.FetchTool;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.json.JSONException;
import org.junit.jupiter.api.Test;
import org.skyscreamer.jsonassert.JSONAssert;

import static io.micronaut.mcp.docs.JsonRpcMessages.EXPECTED_FETCH_TOOL_CALL;
import static io.micronaut.mcp.docs.JsonRpcMessages.FETCH_TOOL_CALL;
import static io.micronaut.mcp.docs.JsonRpcMessages.TOOLS_LIST;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

@Property(name = "micronaut.mcp.server.transport", value = "HTTP")
@Property(name = "spec.name", value = "MicronautModulesFetchTest")
@MicronautTest
class MicronautModulesFetchTest {

    @Test
    void fetchTool(@Client("/") HttpClient httpClient, FetchTool tool) throws JSONException {
        assertEquals("fetch", tool.getName());
        assertEquals("Fetch", tool.getTitle());
        BlockingHttpClient client = httpClient.toBlocking();
        String json = assertDoesNotThrow(() -> client.retrieve(HttpRequest.POST("/mcp", TOOLS_LIST)));
        assertTrue(json.contains("\"name\":\"fetch\""), json);
        String result = assertDoesNotThrow(() -> client.retrieve(HttpRequest.POST("/mcp", FETCH_TOOL_CALL)));
        JSONAssert.assertEquals(EXPECTED_FETCH_TOOL_CALL, result, true);
    }
}

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/tools/fetch/MicronautModulesFetchTest.kt
package io.micronaut.mcp.docs.tools.fetch

import io.micronaut.context.annotation.Property
import io.micronaut.http.HttpRequest
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.mcp.docs.JsonRpcMessages.EXPECTED_FETCH_TOOL_CALL
import io.micronaut.mcp.docs.JsonRpcMessages.FETCH_TOOL_CALL
import io.micronaut.mcp.docs.JsonRpcMessages.TOOLS_LIST
import io.micronaut.mcp.server.tools.fetch.FetchTool
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertDoesNotThrow
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.skyscreamer.jsonassert.JSONAssert

@Property(name = "micronaut.mcp.server.transport", value = "HTTP")
@Property(name = "spec.name", value = "MicronautModulesFetchTest")
@MicronautTest
class MicronautModulesFetchTest {

    @Inject
    @field:Client("/")
    lateinit var httpClient: HttpClient

    @Inject
    lateinit var tool: FetchTool

    @Test
    fun fetchTool() {
        assertEquals("fetch", tool.name)
        assertEquals("Fetch", tool.title)
        val client = httpClient.toBlocking()
        val json = assertDoesNotThrow<String> { client.retrieve(HttpRequest.POST("/mcp", TOOLS_LIST)) }
        assertTrue(json.contains("\"name\":\"fetch\""), json)
        val result = assertDoesNotThrow<String> { client.retrieve(HttpRequest.POST("/mcp", FETCH_TOOL_CALL)) }
        JSONAssert.assertEquals(EXPECTED_FETCH_TOOL_CALL, result, true)
    }
}
